lib/collection/src/shards/shard_holder/resharding.rs

```diff
@@ -284,6 +285,30 @@ impl ShardHolder {
             }
         };
 
+        // Cleanup existing shards if resharding down
+        if is_in_progress && direction == ReshardingDirection::Down {
+            for (&id, shard) in self.shards.iter() {
+                // Skip shards that does not belong to resharding shard key
+                if self.shard_id_to_key_mapping.get(&id) != shard_key.as_ref() {
+                    continue;
+                }
+
+                // Skip target shard
+                if id == shard_id {
+                    continue;
+                }
+
+                // We only cleanup local shards
+                if !shard.is_local().await {
+                    continue;
+                }
+
+                let filter = self.hash_ring_filter(id).expect("hash ring filter");
```

Comment:
Shall we not expect/unwrap here?

Reply:
I think my logic was that we check resharding state right above, and resharding can only be finished/aborted through consensus, and we apply consensus ops sequentially, so it *has to* be available.

Reply:
Fair enough. The `is_in_progress` is good enough for me.

I'm just cautious because I never want to resolve broken consensus again :clown_face: